modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_items.dm

```diff
@@ -0,0 +1,143 @@
+	visible_message(span_notice("[src] beeps: [target_mob]'s mind transfer is now complete."))
+
+	body_component = target_mob.GetComponent(/datum/component/previous_body)
+	scan_body(body_component, user)
+	log_admin("[user] used [src] to put [target_mob]'s mind into a soulcatcher while [target_mob] was still alive.")
+
+	return TRUE
+
+/obj/item/soulcatcher_item/attack_secondary(mob/living/target_mob, mob/living/user, params)
+	if(!target_mob)
+		return FALSE
+
+	if(target_mob.mind || target_mob.ckey || !ishuman(target_mob))
+		to_chat(user, span_warning("[target_mob] is not able to recieve a soul"))
+		return FALSE
+
+	var/list/soul_list = list()
+	for(var/datum/soulcatcher_room/room in linked_soulcatcher.soulcatcher_rooms)
+		for(var/mob/living/soulcatcher_soul/soul in room.current_souls)
+			if(!soul.round_participant || soul.body_scan_needed)
+				continue
+
+			soul_list += soul
+
+	if(length(soul_list) == 0)
+		to_chat(user, span_warning("There are no souls that can be transfered to [target_mob]"))
+		return FALSE
+
+	var/mob/living/soulcatcher_soul/chosen_soul = tgui_input_list(user, "Chose a soul to transfer into the body", name, soul_list)
+	if(!chosen_soul)
+		return FALSE
+
+	if(chosen_soul.previous_body)
+		var/mob/living/old_body = chosen_soul.previous_body.resolve()
+		var/datum/component/previous_body/body_component = old_body.GetComponent(/datum/component/previous_body)
+		body_component.restore_mind = FALSE
+		qdel(body_component)
+
+	chosen_soul.mind.transfer_to(target_mob, TRUE)
+	playsound(src, 'modular_skyrat/modules/modular_implants/sounds/default_good.ogg', 50, FALSE, ignore_walls = FALSE)
+	visible_message(span_notice("[src] beeps: Body transfer complete."))
+	log_admin("[src] was used by [user] to transfer [chosen_soul]'s soulcatcher soul to [target_mob].")
+	qdel(chosen_soul)
+
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/// Attemps to scan the body for the `previous_body component`
+/obj/item/soulcatcher_item/proc/scan_body(datum/component/previous_body/body_to_scan, mob/living/user)
+	var/mob/living/parent_body = body_to_scan.parent
+	var/mob/living/soulcatcher_soul/target_soul = body_to_scan.soulcatcher_soul.resolve()
+	if(!body_to_scan || !target_soul)
+		return FALSE
+
+	if(!target_soul.body_scan_needed)
+		to_chat(user, span_blue("[parent_body] has already been scanned!"))
+		return FALSE
+
+	target_soul.body_scan_needed = FALSE
+	playsound(src, 'modular_skyrat/modules/modular_implants/sounds/default_good.ogg', 50, FALSE, ignore_walls = FALSE)
+	visible_message(span_notice("[src] beeps: [parent_body] is now scanned."))
+
+	to_chat(target_soul, span_blue("Your body has scanned, revealing your true identity."))
+	target_soul.name = parent_body.real_name
+
+	var/datum/preferences/preferences = target_soul.client?.prefs
+	if(preferences)
+		target_soul.soul_desc = preferences.read_preference(/datum/preference/text/flavor_text)
+
+	return TRUE
```

Comment:
This should be a proc on the component, not on the `soulcatcher_item`. (which should also get a name more representative of it's in-game name, or literally anything about it, because this only tells you "yes this is an item related to soulcatcher stuff).